mne/preprocessing/pca_obs.py

```diff
@@ -0,0 +1,314 @@
+                post_range = peak_range
+
+            fitted_art, post_idx_nextPeak = fit_ecg_template(
+                data=data,
+                pca_template=pca_template,
+                a_peak_idx=peak_idx[p],
+                peak_range=peak_range,
+                pre_range=pre_range,
+                post_range=post_range,
+                mid_p=mid_p,
+                fitted_art=fitted_art,
+                post_idx_previous_peak=post_idx_nextPeak,
+                n_samples_fit=n_samples_fit,
+            )
+            # Appending to list instead of using counter
+            window_start_idx.append(peak_idx[p] - peak_range)
+            window_end_idx.append(peak_idx[p] + peak_range)
+
+        # Deals with last edge of data
+        elif p == peak_count - 1:
+            pre_range = math.floor((peak_idx[p] - peak_idx[p - 1]) / 2)
+            post_range = peak_range
+            if pre_range > peak_range:
+                pre_range = peak_range
+            fitted_art, _ = fit_ecg_template(
+                data=data,
+                pca_template=pca_template,
+                a_peak_idx=peak_idx[p],
+                peak_range=peak_range,
+                pre_range=pre_range,
+                post_range=post_range,
+                mid_p=mid_p,
+                fitted_art=fitted_art,
+                post_idx_previous_peak=post_idx_nextPeak,
+                n_samples_fit=n_samples_fit,
+            )
+            window_start_idx.append(peak_idx[p] - peak_range)
+            window_end_idx.append(peak_idx[p] + peak_range)
+
+        # Deals with middle portion of data
+        else:
+            # ---------------- Processing of central data - --------------------
+            # cycle through peak artifacts identified by peakplot
+            pre_range = math.floor((peak_idx[p] - peak_idx[p - 1]) / 2)
+            post_range = math.floor((peak_idx[p + 1] - peak_idx[p]) / 2)
+            if pre_range >= peak_range:
+                pre_range = peak_range
+            if post_range > peak_range:
+                post_range = peak_range
+
+            a_template = pca_template[
+                mid_p - peak_range - 1 : mid_p + peak_range + 1, :
+            ]
+            fitted_art, post_idx_nextPeak = fit_ecg_template(
+                data=data,
+                pca_template=a_template,
+                a_peak_idx=peak_idx[p],
+                peak_range=peak_range,
+                pre_range=pre_range,
+                post_range=post_range,
+                mid_p=mid_p,
+                fitted_art=fitted_art,
+                post_idx_previous_peak=post_idx_nextPeak,
+                n_samples_fit=n_samples_fit,
+            )
+            window_start_idx.append(peak_idx[p] - peak_range)
+            window_end_idx.append(peak_idx[p] + peak_range)
+
+    # Actually subtract the artefact, return needs to be the same shape as input data
+    data -= fitted_art
+    return data
+
+
+def fit_ecg_template(
```

Comment:
Make it clear this isn't a public function
```suggestion
def _fit_ecg_template(
```